agent/connect/ca/provider_vault.go

```diff
@@ -19,7 +21,19 @@ import (
 	"github.com/hashicorp/consul/agent/structs"
 )
 
-const VaultCALeafCertRole = "leaf-cert"
+const (
+	VaultCALeafCertRole = "leaf-cert"
+
+	VaultAuthMethodTypeKubernetes = "kubernetes"
+	VaultAuthMethodTypeLDAP       = "ldap"
+	VaultAuthMethodTypeUserpass   = "userpass"
+	VaultAuthMethodTypeOkta       = "okta"
+	VaultAuthMethodTypeRadius     = "radius"
+	VaultAuthMethodTypeOCI        = "oci"
+	VaultAuthMethodTypeToken      = "token"
```

Comment:
It would have been nice if there was some reflection in the vault api or some sot of enum we could've latched onto here.

Reply:
Unfortunately, they don't have it in the API package.